sources/mma/mma.f90

```diff
@@ -0,0 +1,701 @@
+
+        class(mma_t), intent(inout) :: this
+        integer, intent(in) :: n, m
+        real(kind=rp), intent(in), dimension(n) :: x
+
+
+        this%n = n
+        this%m = m
+
+        if (allocated(this%x)) deallocate(this%x)
+        allocate(this%x(n))
+        this%x = x
+        if (allocated(this%xold1)) deallocate(this%xold1)
+        allocate(this%xold1(n))
+        if (allocated(this%xold2)) deallocate(this%xold2)
+        allocate(this%xold2(n))
+        this%xold1 = this%x
+        this%xold2 = this%x
+
+        if (allocated(this%alpha)) deallocate(this%alpha)
+        allocate(this%alpha(n))
+        if (allocated(this%beta)) deallocate(this%beta)
+        allocate(this%beta(n))
+
+        if (allocated(this%a)) deallocate(this%a) 
+        allocate(this%a(m))
+        if (allocated(this%c)) deallocate(this%c) 
+        allocate(this%c(m))
+        if (allocated(this%d)) deallocate(this%d) 
+        allocate(this%d(m))
+        if (allocated(this%low)) deallocate(this%low) 
+        allocate(this%low(n))
+        if (allocated(this%upp)) deallocate(this%upp) 
+        allocate(this%upp(n))
+        if (allocated(this%xmax)) deallocate(this%xmax) 
+        allocate(this%xmax(n))
+        if (allocated(this%xmin)) deallocate(this%xmin) 
+        allocate(this%xmin(n))
+
+        !internal dummy variables for MMA
+        if (allocated(this%p0j)) deallocate(this%p0j) 
+        allocate(this%p0j(n))
+        if (allocated(this%q0j)) deallocate(this%q0j) 
+        allocate(this%q0j(n))
+        if (allocated(this%pij)) deallocate(this%pij) 
+        allocate(this%pij(m,n))
+        if (allocated(this%qij)) deallocate(this%qij) 
+        allocate(this%qij(m,n))
+        if (allocated(this%bi)) deallocate(this%bi) 
+        allocate(this%bi(m))
+        !---nesessary for KKT check after updating df0dx, fval, dfdx --------
+        if (allocated(this%y)) deallocate(this%y) 
+        allocate(this%y(m))
+        if (allocated(this%lambda)) deallocate(this%lambda) 
+        allocate(this%lambda(m))
+        if (allocated(this%s)) deallocate(this%s) 
+        allocate(this%s(m))
+        if (allocated(this%mu)) deallocate(this%mu) 
+        allocate(this%mu(m))
+        if (allocated(this%xsi)) deallocate(this%xsi) 
+        allocate(this%xsi(n))
+        if (allocated(this%eta)) deallocate(this%eta) 
+        allocate(this%eta(n))
+
+
+        this%epsimin =  1.0e-10 !0.0000001
+        
+        this%max_iter = 100
+
+        !!!!!!! For TEST_CASE_1
+        this%a0=1
+        this%a=0
+        this%c=1000
+        this%d=0
+        !setting the bounds for the design variable based on the problem
+        this%xmax(:) = 10
+        this%xmin(:) = 0 !1e-8 !0.00000001
+
+        !!!!!!! For TEST_CASE_2
+        ! this%a0=1.0
+        ! this%a=0
+        ! this%c=1000
+        ! this%d=1
+        ! !setting the bounds for the design variable based on the problem
+        ! this%xmax(:) = 5
+        ! this%xmin(:) = 0
```

Comment:
This should be inputs to the function. You should never have to modify this file
in order to run it for a different problem.

If you need the posibility of having defaults then look into the use of
`optional` parameters. But remember defaults may never be case specific.